### PR TITLE
アプリケーションの基本構造を作成

### PR DESCRIPTION
**Pull request by BruCandy**
# Summary (GitHub Copilot)

This pull request introduces several new components and functionalities to the application, focusing on improving the user interface and adding CRUD operations for orders, products, and details. The most important changes include the addition of new UI components for different parts of the application, the implementation of CRUD operations using Prisma, and updates to the application's root and index files to enhance the overall structure and functionality.

### New UI Components:

* [`app/components/atoms/button/MenuIconButton.tsx`](diffhunk://#diff-2dd4a30fcf81e9e44e5d44b0de48824bc017858311e8e2022ed720436be7df4eR1-R21): Added a new `MenuIconButton` component using Chakra UI's `IconButton` and `HamburgerIcon` for the menu button.
* [`app/components/molecules/MenueDrawer.tsx`](diffhunk://#diff-2eb7a0b302898379c3492ed3688d538c0ac46ca74090dc0e5567d38ee0dbfa7dR1-R52): Introduced the `MenueDrawer` component, which includes buttons for navigating to different sections of the application.
* [`app/components/organisms/kitchen/OrderCard.tsx`](diffhunk://#diff-0f65c6f780ff0182ac2eaf7ede764ca9ef9ec0d98dc46cb9743d9c612023d408R1-R70): Created the `OrderCard` component to display order details and manage order status transitions.
* [`app/components/organisms/reception/ReceptionCard.tsx`](diffhunk://#diff-e81f1efa3be7a3b807fcec71c7d409b54ffdcc6a204303648caebf48726cdf6aR1-R30): Added the `ReceptionCard` component to handle product orders and cancellations.
* [`app/components/organisms/register/ProductCard.tsx`](diffhunk://#diff-234e741268186c4bd3362f488c89d269d8449cc963ed95a77473651210144f0bR1-R30): Implemented the `ProductCard` component for managing product details, including deletion and modification.

### CRUD Operations:

* [`app/crud/crud_details.ts`](diffhunk://#diff-6f53fb851db978438f1ab7158b80e93cd77baa62d03f4194957ea1c54429985fR1-R29): Added CRUD operations for `order_details`, including creation, reading, and deletion of details.
* [`app/crud/crud_orders.ts`](diffhunk://#diff-6962d4d3077e87833b76289f5319a666b6008948f57cec09718a6423cbb90622R1-R66): Implemented CRUD operations for `orders`, including creation, reading by ID and status, updating status, and deleting orders.
* [`app/crud/crud_products.ts`](diffhunk://#diff-20af0f98be951c6f5bf11293631d15eb0509ddec508aa769bbe0883a65549b51R1-R92): Introduced CRUD operations for `products`, including creation, reading, updating stock, modifying product details, and deletion.

### Application Structure Enhancements:

* [`app/root.tsx`](diffhunk://#diff-4133eb55408b25b35b8e07c696a9dfc97b741c555d7407e8c86d0845e5eecc28R1-L45): Updated the root component to include a `ChakraProvider`, `Header`, and `Sidebar`, and structured the layout using Chakra UI's `Grid` component.
* [`app/routes/_index.tsx`](diffhunk://#diff-8bd8555c577836673951723e36b75ebd6b5ffbb87f0220c01a5b1149c8c7f525L1-L138): Refactored the home page component to use Chakra UI's `Box`, `HStack`, and `VStack` for displaying total sales.
* [`app/routes/header.tsx`](diffhunk://#diff-1d59cdc77af872fc3fa322b7c5534c1fe3063d7428f9df8f46023113035a951dR1-R46): Added a new `Header` component that includes navigation buttons and integrates the `MenuIconButton` and `MenueDrawer` components.

### Additional Changes:

* [`app/hooks/useMessage.ts`](diffhunk://#diff-1ab04a6379a91fcd9465f8c230378c47ae83bb1c5d7a89ad583e3bc0c915b28dR1-R23): Added a custom hook `useMessage` to display toast notifications using Chakra UI's `useToast`.
* [`app/routes/$.tsx`](diffhunk://#diff-6662e418e480ef1a743e5f3f33f1dadb273fbfef3729051b7ee98b65c470de29R1-R4): Added a 404 error page component to handle non-existent routes.

### CODE REVIEW

**Review by r4ai — APPROVED** [review]
LGTM!

**Comment by r4ai** [comment]
@BruCandy 

#4 にてPrettierを導入して自動フォーマットをかけた関係で、一部コンフリクトが発生しているので、コンフリクト解消お願いします！
競合している箇所は、mainブランチの内容を無視して、このブランチの内容で上書きすると良いと思います！

bun.lockbに関しては、package.jsonのコンフリクトを解決した後に、`bun install`を実行してbun.lockbを再生成すると良いと思います

**Comment by BruCandy** [comment]
vscode上でgitignoreにprisma/dev.dbを追加し、MenuIconButton.tsxの内容も簡潔に直したのですが、コンフリクトについてがよくわからないです。本来はコンフリクトが起きていたら、vscode上で以下のようになりますか？
<<<<<<< HEAD

=======

>>>>>>>

File: package.json [hunk 1]
```diff
@@ -35,7 +41,7 @@
     "eslint-plugin-react": "^7.33.2",
     "eslint-plugin-react-hooks": "^4.6.0",
     "postcss": "^8.4.38",
-    "prettier": "^3.3.3",
+    "prisma": "^5.20.0",
```
Comment (3w36zj6) [inline package.json:44]: MUST: Prettierが意図せず削除されてしまっているので修正してください。

**Review by 3w36zj6 — CHANGES_REQUESTED** [review]

File: app/components/molecules/MenueDrawer.tsx
Comment (3w36zj6) [inline app/components/molecules/MenueDrawer.tsx:1]: MUST: 意図しているファイル名は`MenuDrawer.tsx`だと思いますので、修正してください。

**Review by 3w36zj6 — CHANGES_REQUESTED** [review]

File: app/components/molecules/MenueDrawer.tsx
Comment (BruCandy) [reply app/components/molecules/MenueDrawer.tsx:1]: 変更します。反応が遅くなりすみませんでした。

**Review by 3w36zj6 — APPROVED** [review]

**Pull request merged**